src/screens/SearchResultsScreen.tsx

```diff
@@ -24,9 +24,7 @@ export async function loader({ request }: { request: Request }): Promise<string>
 export default function SearchResultsScreen(): JSX.Element {
 	const query: string = useLoaderData() as string;
 	const [movieData, setMovieData] = useState<MovieData | null>(null);
-	const [movieDetails, setMovieDetails] = useState<MovieDetailsData | null>(null);
-
-	/** @TODO Can we combine these useEffects into one and remove the data state? */
+	const [movieDetails, setMovieDetails] = useState<MovieDetailsData[]>([]);
 	useEffect(() => {
```

Comment:
Can we combine these?

```ts
const handler = async () => {
                         const movies = await getMoviesByName(query);
                         setMovieDate(movies)
			if (movies) {
				const movieDetails = await getMovieDetails(movies.results[0].id);
				setMovieDetails(movieDetails);
				for (let i = 0; i < movies.results.length; i++) {
					const movies = await getMovieDetails(movies.results[i].id);
					setMovieDetails((prevArr) => [...prevArr, movies]);
				}
			}
		};

Reply:
Per the comment, I see it doesnt make sense to remove the state altogether, but I think we could remove a useEffect